### ISSUE TITLE
api-docs gives 404

### ISSUE THREAD

**Issue by nickeyzh** [springfox/springfox#2236]
Am using v2.6.1, springboot, java9, and have been able to get swagger-ui.html to return header but there is no content. I have set Docket, resourcehandler, websecurity, but /v2/api-docs is empy. Can i see in the log where it scans classes for methods ? I have set host and pathProvider also.

**Comment by dilipkrish** [comment]
Could you please try with 2.8.0 and report back

**Comment by nickeyzh** [comment]
On 2.8.0 I get unable to infer to base url, which I do not get with 2.7.0. Do you think swagger.io is more stable and better than springfox swagger ?

**Comment by nickeyzh** [comment]
from maven repository, it shows that most uses 2.6.1 and 2.7 and not 2.8... have seen other developers having infer issues with 2.8.0 also..

**Comment by nickeyzh** [comment]
appears to be very little support for this project, and only one person monitoring the forum....

**Comment by dilipkrish** [comment]
@nickeyzh Yes it is for the most part only one person maintaining the project. Without providing details of your project it is impossible to help.

**Comment by nickeyzh** [comment]
Well, its springboot apps, which invokes the Docket bean and configures webSecurity to allow the various urls included, via WebSecurityConfigurerAdapter. Can I see in the logs if it traverses the rest api's to collect info for swagger ?

**Comment by dilipkrish** [comment]
You can turn on logging via normal techiniques... just enable the logging in the packages you need.

**Comment by nickeyzh** [comment]
Have tried, what kind of messages should I e.g. see in the log ?

**Comment by dilipkrish** [comment]
You should see logs like this

```logs

2018-02-17 09:44:11.461  INFO 34680 --- [           main] s.w.s.m.m.a.RequestMappingHandlerMapping : Mapped "{[/swagger-resources]}" onto public org.springframework.http.ResponseEntity<java.util.List<springfox.documentation.swagger.web.SwaggerResource>> springfox.documentation.swagger.web.ApiResourceController.swaggerResources()
2018-02-17 09:44:11.461  INFO 34680 --- [           main] s.w.s.m.m.a.RequestMappingHandlerMapping : Mapped "{[/swagger-resources/configuration/security]}" onto public org.springframework.http.ResponseEntity<springfox.documentation.swagger.web.SecurityConfiguration> springfox.documentation.swagger.web.ApiResourceController.securityConfiguration()
2018-02-17 09:44:11.462  INFO 34680 --- [           main] s.w.s.m.m.a.RequestMappingHandlerMapping : Mapped "{[/swagger-resources/configuration/ui]}" onto public org.springframework.http.ResponseEntity<springfox.documentation.swagger.web.UiConfiguration> springfox.documentation.swagger.web.ApiResourceController.uiConfiguration()
2018-02-17 09:44:11.465  INFO 34680 --- [           main] s.w.s.m.m.a.RequestMappingHandlerMapping : Mapped "{[/error]}" onto public org.springframework.http.ResponseEntity<java.util.Map<java.lang.String, java.lang.Object>> org.springframework.boot.autoconfigure.web.BasicErrorController.error(javax.servlet.http.HttpServletRequest)
2018-02-17 09:44:11.465  INFO 34680 --- [           main] s.w.s.m.m.a.RequestMappingHandlerMapping : Mapped "{[/error],produces=[text/html]}" onto public org.springframework.web.servlet.ModelAndView org.springframework.boot.autoconfigure.web.BasicErrorController.errorHtml(javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse)
2018-02-17 09:44:11.591  INFO 34680 --- [           main] pertySourcedRequestMappingHandlerMapping : Mapped URL path [/api-docs/{swaggerGroup}/{apiDeclaration}] onto method [public org.springframework.http.ResponseEntity<springfox.documentation.spring.web.json.Json> springfox.documentation.swagger1.web.Swagger1Controller.getApiListing(java.lang.String,java.lang.String,javax.servlet.http.HttpServletRequest)]
2018-02-17 09:44:11.592  INFO 34680 --- [           main] pertySourcedRequestMappingHandlerMapping : Mapped URL path [/api-docs] onto method [public org.springframework.http.ResponseEntity<springfox.documentation.spring.web.json.Json> springfox.documentation.swagger1.web.Swagger1Controller.getResourceListing(java.lang.String)]
2018-02-17 09:44:11.596  INFO 34680 --- [           main] pertySourcedRequestMappingHandlerMapping : Mapped URL path [/v2/api-docs] onto method [public org.springframework.http.ResponseEntity<springfox.documentation.spring.web.json.Json> springfox.documentation.swagger2.web.Swagger2Controller.getDocumentation(java.lang.String,javax.servlet.http.HttpServletRequest)]

```

Make sure you have logging dialed up to INFO least for `org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerMapping ` and `springfox.documentation.spring.web.PropertySourcedRequestMappingHandlerMapping`

**Comment by dilipkrish** [comment]
Seems like its all mapped as expected. The last line shows its mapped url to `/v2/api-docs`

**Comment by nickeyzh** [comment]
I have this in access log:

"GET /webSelv/null/swagger-resources/configuration/ui HTTP/1.1" 404 534

Any idea ?

**Comment by nickeyzh** [comment]
What is alternative to springfox ?

**Comment by nickeyzh** [comment]
Error:

16:39:34.064 [localhost-startStop-3] DEBUG o.s.jndi.JndiLocatorDelegate 
- Converted JNDI name 
[java:comp/env/springfox.documentation.swagger.v2.host] not found - 
trying original name [springfox.documentation.swagger.v2.host]. 
javax.naming.NameNotFoundException: Name 
[springfox.documentation.swagger.v2.host] is not bound in this Context. 
Unable to find [springfox.documentation.swagger.v2.host].
16:39:34.064 [localhost-startStop-3] DEBUG o.s.jndi.JndiPropertySource - 
JNDI lookup for name [springfox.documentation.swagger.v2.host] threw 
NamingException with message: Name 
[springfox.documentation.swagger.v2.host] is not bound in this Context. 
Unable to find [springfox.documentation.swagger.v2.host].. Returning null.
16:39:34.064 [localhost-startStop-3] DEBUG 
o.s.c.e.PropertySourcesPropertyResolver - Could not find key 
'springfox.documentation.swagger.v2.host' in any property source
16:39:34.064 [localhost-startStop-3] DEBUG o.s.jndi.JndiLocatorDelegate 
- Converted JNDI name 
[java:comp/env/springfox.documentation.swagger.v2.path] not found - 
trying original name [springfox.documentation.swagger.v2.path]. 
javax.naming.NameNotFoundException: Name 
[springfox.documentation.swagger.v2.path] is not bound in this Context. 
Unable to find [springfox.documentation.swagger.v2.path].

should host and path not be set ?

**Comment by nickeyzh** [comment]
Not really working, springfox:

"GET /webSelv/v2/api-docs HTTP/1.1" 404 534

Dispite:

Mapped URL path [/v2/api-docs] onto method [public 
org.springframework.http.ResponseEntity<springfox.documentation.spring.web.json.Json> 
springfox.documentation.swagger2.web.Swagger2Controller.getDocumentation(java.lang.String,javax.servlet.http.HttpServletRequest)]

and websecurity disabled:

Creating filter chain: Ant [pattern='/v2/api-docs/**'], []

What to try next, already on 2.8 - maybe 2.9 ?

**Comment by dilipkrish** [comment]
Hard to say without a reporducable project or test. Could you perhaps clone [the demos](https://github.com/springfox/springfox-demos) application and try and reproduce the problem. Otherwise see how your project is different from the demos.

**Comment by nickeyzh** [comment]
you can improve transparency, as many others have been commenting on the 
issues with springfox, so more logging please. When is next release ?

**Comment by nickeyzh** [comment]
http://localhost:8081/base/v2/api-docs gives 404 on this in access 
/base/null/swagger-resources/configuration/ui - where does the null come 
from ? bug ?

or this NullPointerException:

22:10:42.304 [localhost-startStop-1] DEBUG 
o.s.b.f.annotation.InjectionMetadata - Processing injected element of 
bean 'serviceModelToSwagger2MapperImpl': AutowiredFieldElement for 
private springfox.documentation.swagger2.mappers.ModelMapper 
springfox.documentation.swagger2.mappers.ServiceModelToSwagger2MapperImpl.modelMapper
22:10:42.304 [localhost-startStop-1] DEBUG 
o.s.core.annotation.AnnotationUtils - Failed to meta-introspect 
annotation [interface 
org.springframework.beans.factory.annotation.Autowired]: 
java.lang.NullPointerException
22:10:42.305 [localhost-startStop-1] DEBUG 
o.s.b.f.s.DefaultListableBeanFactory - Returning cached instance of 
singleton bean 'modelMapperImpl'
22:10:42.316 [localhost-startStop-1] DEBUG 
o.s.b.f.a.AutowiredAnnotationBeanPostProcessor - Autowiring by type from 
bean name 'serviceModelToSwagger2MapperImpl' to bean named 'modelMapperImpl'
22:10:42.316 [localhost-startStop-1] DEBUG 
o.s.b.f.annotation.InjectionMetadata - Processing injected element of 
bean 'serviceModelToSwagger2MapperImpl': AutowiredFieldElement for 
private springfox.documentation.swagger2.mappers.ParameterMapper 
springfox.documentation.swagger2.mappers.ServiceModelToSwagger2MapperImpl.parameterMapper
22:10:42.316 [localhost-startStop-1] DEBUG 
o.s.core.annotation.AnnotationUtils - Failed to meta-introspect 
annotation [interface 
org.springframework.beans.factory.annotation.Autowired]: 
java.lang.NullPointerException
